cmd/otel-allocator/prehook/relabel.go

```diff
@@ -58,13 +44,11 @@ func (tf *RelabelConfigTargetFilter) Apply(targets map[string]*target.Item) map[
 	// Note: jobNameKey != tItem.JobName (jobNameKey is hashed)
 	for jobNameKey, tItem := range targets {
 		keepTarget := true
-		lset := convertLabelToPromLabelSet(tItem.Labels)
+		lset := tItem.Labels
 		for _, cfg := range tf.relabelCfg[tItem.JobName] {
-			if newLset, keep := relabel.Process(lset, cfg); !keep {
-				keepTarget = false
+			lset, keepTarget = relabel.Process(lset, cfg)
+			if !keepTarget {
```

Comment:
I'm wondering if we could delete the job on the first if !keepTarget

Reply:
On the first !keepTarget, we break out of the loop, and then delete. So I think we're doing what you're suggesting?

Really grateful for your review on these kinds of changes by the way!

Reply:
I'm more than happy to help somehow I can 🙇🏽‍♂️ 
I was thinking about delete inside the if we break, does it makes sense?